bokeh/charts/utils.py

```diff
@@ -38,3 +44,109 @@ def polar_to_cartesian(r, start_angles, end_angles):
         points.append(cartesian(r, (end + start)/2))
 
     return zip(*points)
+
+# TODO: Experimental implementation. This should really be a shared
+#       pattern between plotting/charts and other bokeh interfaces
+class Figure(object):
+    def __init__(self, *charts, **kwargs):
+        self.filename = kwargs.pop('filename', None)
+        self.server = kwargs.pop('server', None)
+        self.notebook = kwargs.pop('notebook', None)
+        self.title = kwargs.pop('title', '')
+        self.children = kwargs.pop('children', None)
+        self.charts = charts
+        self.doc = Document()
+        self.doc.hold(True)
+
+        if self.server:
+            self.session = Session()
+            self.session.use_doc(self.server)
+            self.session.load_document(self.doc)
+
+        if self.children:
+            from bokeh.models import VBox
+            self.doc.add(VBox(children=self.children))
+
+        self.plot = None
+        xdr, ydr = None, None
+        for i, chart in enumerate(self.charts):
+            chart.doc = self.doc
+            if self.server:
+                chart.session = self.session
+
+            # Force the chart to create the underlying plot
+            chart._setup_show()
+            chart._prepare_show()
+            chart._show_teardown()
+
+            if not self.title:
+                self.title = chart.chart.title
+
+        # reset the pot title with the one set for the Figure
+        self.doc._current_plot.title = self.title
+
+    def show(self):
```

Comment:
code duplication... I know that this need to be shared across multiple places, just point it out...